### PR TITLE
win midi: refactor pause state booleans into enum

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: src/i_winmusic.c [hunk 1]
```diff
@@ -1534,7 +1539,7 @@ static void I_WIN_PauseSong(void *handle)
         return;
     }
 
-    paused = true;
+    music_state = stop_sound;
```
Comment (ceski-1) [inline src/i_winmusic.c:1542]: If the main thread can modify `music_state` before initial playback has finished or whenever it's inside the switch statement, then there's no way to ensure the correct order of execution. You can try to guard with `if (music_state != initial_playback)` inside the pause and resume functions but that's still not a complete solution.

Comment (rfomin) [reply src/i_winmusic.c]: Okay, how about just pause states? https://github.com/fabiangreffrath/woof/pull/1114/commits/bc44667c6794fd7c021fb47e3b403f0100423e70
BTW, ZMusic has something similar: https://github.com/ZDoom/ZMusic/blob/fc30a8f1e4350dba68e23bb702dfa24df3fe525e/source/zmusic/musinfo.h#L36-L41

Comment (ceski-1) [reply src/i_winmusic.c]: Looks good. Let's flush that buffer immediately to be safe.
```diff
    if (music_state == stopped)
    {
        StopSound();
+       StreamOut();
        music_state = paused;
+       return;
    }
```

Comment (rfomin) [reply src/i_winmusic.c]: Done in https://github.com/fabiangreffrath/woof/pull/1114/commits/c1a772c20c805bedd446ebea30967aa42a9132a3

**Comment by fabiangreffrath** [comment]
Is this change set already on par with Choco?

**Comment by rfomin** [comment]
> Is this change set already on par with Choco?

It seems not, I will update it.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**